libsrc/c64/crt0.s

```diff
@@ -98,14 +85,51 @@ L2:     lda     zpsave,x
 
         rts
 
+
 ; ------------------------------------------------------------------------
-; Data
 
-.segment        "ZPSAVE"
+.segment        "INIT"
 
-zpsave: .res    zpspace
+initstart:
 
-.bss
+; Save the zero-page locations that we need.
+
+        ldx     #zpspace-1
+L1:     lda     sp,x
+        sta     zpsave,x
+        dex
+        bpl     L1
+
+; Clear the BSS data.
+
+        jsr     zerobss
+
+; Set up the stack.
+
+        lda     #<(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
+        ldx     #>(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
+        sta     sp
+        stx     sp+1            ; Set argument stack ptr
+
+; Call the module constructors.
+
+        jsr     initlib
+
+; Push the command-line arguments; and, call main().
+
+        jmp     callmain
+
+
+; ------------------------------------------------------------------------
+; Data
+
+.data
 
-spsave: .res    1
 mmusave:.res    1
+spsave: .res    1
```

Comment:
Okay, trying to sort this out...

> The .res in the third line is a sign to readers that the initial values in those locations are irrelevant.

Sorry, but I just can't follow you here!

The semantics of variables not relying on implicit initialization either by being loaded from disk or via zerobss() just isn't something usually expressed in neither assembly nor C. While I can see that it might make sense to have ways to express that...
* I don't see us changing the language we're working with. After all this is a community project and others should be at least be able to read the source code.
* According to http://cc65.github.io/doc/ca65.html#ss11.91 `.res` by no means results in the variable having an undefined initial value. Rather it's initialized with zero. So even someone would try to understand what you want to express he'd have no chance doing so when reading the `.res` docs.

Result: The desired cc65 style is to use `.res` in the BSS and `.byte` (and friends) for DATA and RODATA. I'd appreciate if you would adjust your code.

> `argv` is defined as [...]

What is the reason to not place it in the BSS - with the obvious benefit of reducing the file size?

In contrast `mmusave` and `spsave` are from a semantic perspective BSS type variables but have to live somewhere else for technical reasons - they are used before BSS is cleared. With that perspective in mind I can see the point of `.res` for them but I'd vote for a comment stating why they can't live in BSS. After all it's obviously not that obvious ;-)